packages/teleport/src/Sessions/SessionList/SessionJoinBtn.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+
+import { ButtonBorder, Text, Box, Menu, MenuItem } from 'design';
+import { CarrotDown } from 'design/Icon';
+
+import cfg from 'teleport/config';
+import { ParticipantMode } from 'teleport/services/session';
+
+export const SessionJoinBtn = ({
+  sid,
+  clusterId,
+  participantModes,
+}: {
+  sid: string;
+  clusterId: string;
+  participantModes: ParticipantMode[];
+}) => {
+  // Sorts the list of participantModes so that they are consistently shown in the order of "observer" -> "moderator" -> "peer"
```

Comment:
What do you think about something like this instead?

```
const modes = {
  'observer': 1,
  'moderator': 2,
  'peer': 3,

const sortedModes = participantModes.sort((a,b) => modes[a] - modes[b]))
```

This way, it's obvious that `observer` has highest priority because it has the lowest number.